src/fsharp/TastOps.fs

```diff
@@ -1807,11 +1805,47 @@ let rec isUnmanagedTy g ty =
                 true
             elif tycon.IsStructOrEnumTycon then
                 match tycon.TyparsNoRange with
-                | [] -> tycon.AllInstanceFieldsAsList |> List.forall (fun r -> isUnmanagedTy g r.rfield_type) 
-                | _ -> false // generic structs are never 
+                | [] -> tycon.AllInstanceFieldsAsList |> List.forall (fun r -> isUnmanagedTy g r.rfield_type)
+                | typars ->
+                    // Handle generic structs
+                    // REVIEW: This may not be the most optimal, but it's probably better than having to iterate over all type arguments for every field.
+                    //     However, what we have currently is probably just fine as unmanaged constraints are used infrequently; even more so when combined with generic struct construction.
+                    let lookup = Dictionary(typars.Length)
```

Comment:
The code is a little odd.  Normally we would not enforce the transitive constraints like this - we would require that each type parameter itself has the `unmanaged` constraint, e.g. consider this:

```
    [<Struct>]
    type C1<'T when 'T : unmanaged> = 
        ...

    [<Struct>]
    type C2<'U when 'U : unmanaged> = 
        ... field C1<'U> ....
```

Then when checking whether `C2<int>` is unmanaged we would check whether the instantiated field type `C1<int>` is unmanaged.  It looks like you're re-implementing substitution here.

What if you use simply this:
```
            elif tycon.IsStructOrEnumTycon then
                let tinst = mkInstForAppTy g ty
                tycon.AllInstanceFieldsAsList |> List.forall (fun r -> isUnmanagedTy g (actualTyOfRecdField tinst r))
```

Reply:
Note that `actualTyOfRecdField` gets the instantiated field type under a governing instantiation derived from `ty`

Reply:
`actualTyOfRecdField` is exactly what I need and will simplify the existing code; I was reimplementing substitution. I had a hard time trying to find such a function before.